gem/test/samson_secret_puller_test.rb

```diff
@@ -154,6 +154,17 @@ def change_constant(constant_name, value)
     end
   end
 
+  describe '.dup' do
+    it "generates a complete hash" do
+      SamsonSecretPuller.dup["FOO"].must_equal "bar"
+    end
+
+    it "generates a copy" do
+      SamsonSecretPuller.dup["FOO"] = "baz"
+      SamsonSecretPuller.dup["FOO"].must_equal "bar"
+    end
```

Comment:
the real test would be if we can do a dup - change - replace and get back the original state ?